homeassistant/components/device_tracker/snmp.py

```diff
@@ -94,8 +113,12 @@ def get_snmp_data(self):
         """Fetch MAC addresses from access point via SNMP."""
         devices = []
 
-        errindication, errstatus, errindex, restable = self.snmp.nextCmd(
-            self.community, self.host, self.baseoid)
+        if self.version == 3:
```

Comment:
If you use the same variable above, say `self.request`, instead of `self.community` and `self.userdata` then you don't require this if statement. You also don't need the `self.version`